### ISSUE TITLE
Docker build hangs "Waiting for exit code to appear for remote command..."

### ISSUE THREAD

**Issue by russroy** [mitchellh/packer#901]
My docker build hangs with 

```
....
==> docker: Provisioning with chef-solo
2014/02/21 08:28:31 ui: ==> docker: Provisioning with chef-solo
    docker: Installing Chef...
2014/02/21 08:28:31 ui:     docker: Installing Chef...
2014/02/21 08:28:31 /Users/rroy/0.5.1_darwin_amd64/packer-builder-docker: 2014/02/21 08:28:31 Executing in container 73f2961fee6988f4f2da2d3328a04bdb73d5e240704ec6f1af58cd72576f24a5: "(curl -L https://www.opscode.com/chef/install.sh | sudo bash) >/packer-files/cmd870430499 2>&1; echo $? >/packer-files/cmd870430499-exit"
2014/02/21 08:28:33 /Users/rroy/0.5.1_darwin_amd64/packer-builder-docker: 2014/02/21 08:28:33 Waiting for exit code to appear for remote command...
```

When I attach to this running container from a separate terminal window I can see that the file that it's waiting on does in fact exist with content "0":
```
rroy:coreos-vagrant rroy$ docker attach compassionate_franklin
root@73f2961fee69:/# ls /packer-files/
cmd870430499  cmd870430499-exit
root@73f2961fee69:/# date
Thu Feb 20 23:07:02 UTC 2014
root@73f2961fee69:/# cd /packer-files/
root@73f2961fee69:/packer-files# ls -lrt
total 8
-rw-r--r-- 1 root root 1332 Feb 20 22:56 cmd870430499
-rw-r--r-- 1 root root    2 Feb 20 22:56 cmd870430499-exit
root@73f2961fee69:/packer-files# cat cmd870430499
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 13347  100 13347    0     0  66249      0 --:--:-- --:--:-- --:--:-- 91417
Downloading Chef  for ubuntu...
downloading https://www.opscode.com/chef/metadata?v=&prerelease=false&p=ubuntu&pv=12.04&m=x86_64
  to file /tmp/install.sh.11/metadata.txt
trying curl...
url	https://opscode-omnibus-packages.s3.amazonaws.com/ubuntu/12.04/x86_64/chef_11.10.4-1.ubuntu.12.04_amd64.deb
md5	67ebae37bcd30b34ab2242727d7b442e
sha256	4da16f60e20c7499748b1ea1b93fa174a13422eeb7c141370c83c86ba9c8b032
downloaded metadata file looks valid...
downloading https://opscode-omnibus-packages.s3.amazonaws.com/ubuntu/12.04/x86_64/chef_11.10.4-1.ubuntu.12.04_amd64.deb
  to file /tmp/install.sh.11/chef_11.10.4-1.ubuntu.12.04_amd64.deb
trying curl...
Checksum compare with sha256sum succeeded.
Installing Chef 
installing with dpkg...
(Reading database ... 29011 files and directories currently installed.)
Preparing to replace chef 11.10.0-1.ubuntu.12.04 (using .../chef_11.10.4-1.ubuntu.12.04_amd64.deb) ...
Unpacking replacement chef ...
Setting up chef (11.10.4-1.ubuntu.12.04) ...
Thank you for installing Chef!
root@73f2961fee69:/packer-files# cat cmd870430499-exit
0
root@73f2961fee69:/packer-files# 
```
Using docker 0.8.0, packer 0.5.1.

My template:
```
{
   "builders": [{
  	"type": "docker",
	"image": "linux/chef-solo",
	"export_path": "image.tar"
    }],
    "provisioners": [{
  	"type": "chef-solo",
  	"cookbook_paths": ["./chef-repo/cookbooks/"],
	"run_list" : ["jenkins::master"]
    }]
}
```

**Comment by CambridgeMBN** [comment]
Docker builds also hang for me when provisioning. I'm using an Ubuntu docker image running on CoreOS, with a docker client binary on Ubuntu linked to the CoreOS docker socket.

This occurs even though I've mounted the tmp directory on Ubuntu.

**Comment by xmik** [comment]
@CambridgeMBN, in #940 @mitchellh wrote "The docker builder only works if Docker is on the same host". I verified with Packer 0.7.5 and also with Packer commit https://github.com/mitchellh/packer/commit/edf3415c6e2248c8096f9c16cbad5022b73d6f21. It hangs with "waiting for exit code to appear for remote command...".

**Comment by CambridgeMBN** [comment]
@xmik I've successfully managed to use the docker builder on a remote host with the [patch] (https://github.com/stefancocora/packer-issue1752-fix) made by @stefancocora in #1752.

**Comment by xmik** [comment]
@CambridgeMBN, I actually tested this patch too (but didn't know about the fork) and it hanged with the same message when using remote host. (However that solved #1752 for me on local Docker host). 

That patch (slightly changed) was merged into https://github.com/mitchellh/packer/commit/edf3415c6e2248c8096f9c16cbad5022b73d6f21.

**Comment by nathan-ms** [comment]
One possible workaround is to change the location of `TMPDIR` to somewhere in `/Users`. If you are using boot2docker on Mac then `/Users` should be mounted automatically. I completed a packer run after `export TMPDIR=~/tmp` and everything worked as it should.

**Comment by narqo** [comment]
@nathan-ms the workaround with `TMPDIR` env variable is really work. Thanks!

**Comment by raphaelmerx** [comment]
Thanks @nathan-ms, setting the `TMPDIR` environment variable worked like a charm.

**Comment by harelba** [comment]
TMPDIR workaround works for me as well! Thanks
